### PR TITLE
MGMT-16002: Accept *yml.patch* and *yaml.patch* extensions

### PR DESCRIPTION
**Pull request by paul-maidment**
Presently we do not accept filenames containing *yml.patch* or *yaml.patch*
We should accept these files.
This PR rectifies that.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
Performed curl requests against an assisted-test-infra based cluster that is running this service image. *yml.patch* and *yaml.patch* files were accepted as expected.
- [x] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: data/default_release_images.json [hunk 1]
```diff
@@ -92,17 +92,17 @@
         "cpu_architectures": [
             "x86_64"
         ],
-        "url": "quay.io/openshift-release-dev/ocp-release:4.13.35-x86_64",
-        "version": "4.13.35"
+        "url": "quay.io/openshift-release-dev/ocp-release:4.13.36-x86_64",
```
Comment (gamli75) [inline data/default_release_images.json:95]: all changes in this file are not related to your PR - please remove it

File: deploy/podman/configmap.yml [hunk 2]
```diff
@@ -27,7 +27,7 @@ data:
...
...ay.io/openshift-release-dev/ocp-release:4.13.36-multi","version":"4.13.36-multi"},{"openshift_version":"4.14","cpu_architecture":"x86_64","cpu_architectures":["x86_64"],"url":"quay.io/openshift-release-dev/ocp-release:4.14.15-x86_64","version":"4.14.15"},{"openshift_version":"4.14","cpu_architecture":"arm64","cpu_architectures":["arm64"],"url":"quay.io/openshift-release-dev/ocp-release:4.14.15-aarch64","version":"4.14.15"},{"openshift_version":"4.14-multi","cpu_architecture":"multi","cpu_architectures":["x86_64","arm64","ppc64le","s390x"],"url":"quay.io/openshift-release-dev/ocp-release:4.14.15-multi","version":"4.14.15-multi"},{"openshift_version":"4.15","cpu_architecture":"x86_64","cpu_architectures":["x86_64"],"url":"quay.io/openshift-release-dev/ocp-release:4.15.0-x86_64","version":"4.15.0","default":true},{"openshift_version":"4.15","cpu_architecture":"arm64","cpu_architectures":["arm64"],"url":"quay.io/openshift-release-dev/ocp-release:4.15.0-aarch64","version":"4.15.0"},{"openshift_version":"4.15-multi","cpu_architecture":"multi","cpu_architectures":["x86_64","arm64","ppc64le","s390x"],"url":"quay.io/openshift-release-dev/ocp-release:4.15.0-multi","version":"4.15.0-multi"}]'
```
Comment (gamli75) [inline deploy/podman/configmap.yml:30]: all changes in this file are not related to your PR - please remove it

File: openshift/template.yaml [hunk 3]
```diff
@@ -17,7 +17,7 @@ parameters:
...
...elease-dev/ocp-release:4.13.36-multi","version":"4.13.36-multi"},{"openshift_version":"4.14","cpu_architecture":"x86_64","cpu_architectures":["x86_64"],"url":"quay.io/openshift-release-dev/ocp-release:4.14.15-x86_64","version":"4.14.15"},{"openshift_version":"4.14","cpu_architecture":"arm64","cpu_architectures":["arm64"],"url":"quay.io/openshift-release-dev/ocp-release:4.14.15-aarch64","version":"4.14.15"},{"openshift_version":"4.14-multi","cpu_architecture":"multi","cpu_architectures":["x86_64","arm64","ppc64le","s390x"],"url":"quay.io/openshift-release-dev/ocp-release:4.14.15-multi","version":"4.14.15-multi"},{"openshift_version":"4.15","cpu_architecture":"x86_64","cpu_architectures":["x86_64"],"url":"quay.io/openshift-release-dev/ocp-release:4.15.0-x86_64","version":"4.15.0","default":true},{"openshift_version":"4.15","cpu_architecture":"arm64","cpu_architectures":["arm64"],"url":"quay.io/openshift-release-dev/ocp-release:4.15.0-aarch64","version":"4.15.0"},{"openshift_version":"4.15-multi","cpu_architecture":"multi","cpu_architectures":["x86_64","arm64","ppc64le","s390x"],"url":"quay.io/openshift-release-dev/ocp-release:4.15.0-multi","version":"4.15.0-multi"}]' # release images
```
Comment (gamli75) [inline openshift/template.yaml:20]: all changes in this file are not related to your PR - please remove it

**Comment by gamli75** [comment]
@adriengentil please review

**Comment by adriengentil** [comment]
@paul-maidment can you add a word about the patch extension in the doc? https://github.com/openshift/assisted-service/blob/master/docs/user-guide/install-customization.md?plain=1#L9

File: internal/manifests/manifests.go [hunk 4]
```diff
@@ -540,12 +540,12 @@ func (m *Manifests) validateUserSuppliedManifest(ctx context.Context, clusterID
 		if !json.Valid(manifestContent) {
 			return m.prepareAndLogError(ctx, http.StatusBadRequest, errors.Errorf("Manifest content of file %s for cluster ID %s has an illegal JSON format", fileName, string(clusterID)))
 		}
-	} else if strings.HasPrefix(extension, ".patch") && (strings.Contains(fileName, ".yaml.patch") || strings.Contains(fileName, ".yml.patch")) {
+	} else if strings.Contains(fileName, ".yaml.patch") || strings.Contains(fileName, ".yml.patch") {
```
Comment (adriengentil) [inline internal/manifests/manifests.go:543]: just to make sure it's intended,  we accept `myfile.yaml.patch` and `myfile.yaml.patchfoobar`, right?

File: internal/manifests/manifests.go [hunk 5]
```diff
@@ -540,12 +540,12 @@ func (m *Manifests) validateUserSuppliedManifest(ctx context.Context, clusterID
...
 		if err := isValidYaml(manifestContent); err != nil {
```
Comment (adriengentil) [inline internal/manifests/manifests.go:544]: shouldn't we validate that the file is a valid yaml patch instead? https://github.com/openshift/assisted-service/blob/master/internal/common/common.go#L482

File: internal/manifests/manifests.go (hunk 4 above)
Comment (paul-maidment) [reply internal/manifests/manifests.go:543]: Ideally it should be `myfile.yaml.patch` or `myfile.yaml.patch.foobar`, I'll fix things to account for the extra separator.

Comment (paul-maidment) [reply internal/manifests/manifests.go:543]: But yes, any filename containing `.yaml.patch.` or `.yml.patch.` should be permitted

File: internal/manifests/manifests.go (hunk 5 above)
Comment (paul-maidment) [reply internal/manifests/manifests.go]: Agreed!
I have changed this.

**Review by adriengentil — APPROVED** [review]
